### DISCUSSION TITLE
vcpkg QtWebEngine

### DISCUSSION THREAD

**Discussion by Melio-1** [microsoft/vcpkg#34583] [Q&A] [answered]
Why QtWebEngine take too long time to build dbg and rel?

and if my laptop shut down and i wnat install it again it's start from begining even i use --editable 

Note: i use this command { .\vcpkg.exe install qtwebengine:x64-windows --x-buildtrees-root=P:\dev --editable }

**Comment by dg0yt** [comment]
qwebengine is not a toy project. 

`--editable` is about not cleaning the sources, so modifications are possible. It will still start with a clean build directory. It is not meant to safe build time.

**Reply by Melio-1** [reply]
how can i save build time i'm in area that electracity shut down about 4 hrs and trun 2 hrs

my laptop shutting down in about 3.5 hrs

yeah and how to make QtWebEngine in shorter path during qt install

**Reply by Neumann-A** [reply]
Your laptop is probably not being powerful enough to build qtwebengine in a timely manner you could try x64-windows-release if release libs are enough for you.

**Comment by Melio-1** [comment]
> Your laptop is probably not being powerful enough to build qtwebengine in a timely manner you could try x64-windows-release if release libs are enough for you.

I've ASUS TUF A15 (2023) with 32 GB RAM (5600) and 2TB Samsung PCIe gen 3

and i want to install qt library only but vcpkg froce my to install qtwebengine dependencies

**Reply by Neumann-A** [reply, marked as answer]
`Elapsed time to handle qtwebengine:x64-windows-release: 1.1 h` and that is on an `AMD Ryzen 9 7900X 12-Core` CI takes 2.2 h for both configs.  You have an `AMD Ryzen™ 7 6800H Mobile Processor (8-core/16-thread, 20MB cache, up to 4.7 GHz max boost)` which probably looses boost after like 5 min. So you are probably around 3.8 GHz which is 80% of my speed. Then you have only 66% of the cores. So single config builds should take give or take 2.1 h to 3h for you. Double that if you build both configs.

**Reply by Melio-1** [reply]
No i've AMD Ryzen™ 7 7735H Mobile Processor (8-core/16-thread, 20MB cache, up to 4.75 GHz max boost) 

when electracity turn i change the mode into turbo mode so increases to < 4.4 GHz 

it's take 12h because i trun battery saver which decrease cores speed to 3.2 GHz and some times the laptop going to sleep mode so that's why taking this hole time

finally install qt library 😉

**Comment by Melio-1** [comment]
> `Elapsed time to handle qtwebengine:x64-windows-release: 1.1 h` and that is on an `AMD Ryzen 9 7900X 12-Core` CI takes 2.2 h for both configs. You have an `AMD Ryzen™ 7 6800H Mobile Processor (8-core/16-thread, 20MB cache, up to 4.7 GHz max boost)` which probably looses boost after like 5 min. So you are probably around 3.8 GHz which is 80% of my speed. Then you have only 66% of the cores. So single config builds should take give or take 2.1 h to 3h for you. Double that if you build both configs.



> `Elapsed time to handle qtwebengine:x64-windows-release: 1.1 h` and that is on an `AMD Ryzen 9 7900X 12-Core` CI takes 2.2 h for both configs. You have an `AMD Ryzen™ 7 6800H Mobile Processor (8-core/16-thread, 20MB cache, up to 4.7 GHz max boost)` which probably looses boost after like 5 min. So you are probably around 3.8 GHz which is 80% of my speed. Then you have only 66% of the cores. So single config builds should take give or take 2.1 h to 3h for you. Double that if you build both configs.
